plugins/pulp_rpm/plugins/distributors/yum/configuration.py

```diff
@@ -393,10 +394,15 @@ def get_repo_checksum_type(publish_conduit, config):
 def get_gpg_sign_options(repo=None, config=None):
     cfg = config or {}
     cmd = cfg.get(GPG_CMD)
-    if not cmd:
-        # Default to plain gpg
-        cmd = '/usr/bin/gpg --yes --detach-sign --armor'
     key_id = cfg.get(GPG_KEY_ID)
+    if cmd:
+        cmd = shlex.split(cmd)
+    else:
+        # Default to plain gpg
+        if key_id:
```

Comment:
You could set cmd = ['/usr/bin/gpg', '--yes', '--detach-sign', '--armor'] outside of the if, and if key_id, extend it with ['--default-key', key_id]

Reply:
There is a test in test_util that would break with your change, it needs to be updated.
plugins/test/unit/plugins/distributors/yum/test_publish.py has one test, I'd really advocate for adding a test for no command but key present. I know it sucks to delay a patch because of unit tests, but in my experience it pays off later.

Reply:
Actually plugins/test/unit/plugins/distributors/yum/test_configuration.py has the place where this may be easier to test.